stefaniaPopescu26/assignment3/WordSearch.java

```diff
@@ -0,0 +1,148 @@
+import javafx.util.Pair;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+import java.util.Vector;
+
+/*
+    The code reads the input from two files:
+
+    wordSearch.txt :
+        row_number col_number
+        // the grid
+
+    dictionary.txt :
+        number_of_words
+        word_1
+        .....
+        word_n
+*/
+
+public class WordSearch {
+    private Dictionary dictionary;
+    private String[][] grid;
+    private int rows;
+    private int cols;
+
+    public WordSearch(String word_filename, String dictionary_filename) {
+        readGrid(word_filename);
+        dictionary = new Dictionary(dictionary_filename);
+    }
+
+    public void setGrid(int rows, int cols, String[][] grid) {
+        this.rows = rows;
+        this.cols = cols;
+        this.grid = grid;
+    }
+
+    public void delDictionary() {
+        dictionary.clear();
+    }
+
+    private void readGrid(String filename) {
+        try {
+            Scanner sc = new Scanner(new File(filename));
+
+            this.rows = sc.nextInt();
+            this.cols = sc.nextInt();
+            grid = new String[rows][cols];
+
+            if (rows != 0 && cols != 0)
+                sc.nextLine();
+
+            for(int row_index = 0; row_index < rows; row_index++) {
+                for (int col_index = 0; col_index < cols; col_index++) {
+                        grid[row_index][col_index] = sc.next();
+                }
+                if (row_index != rows - 1)
+                    sc.nextLine();
+            }
+
+            sc.close();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /*
+        This method returns the vector of all valid neighbours for a cell using
+        two arrays of offsets, for rows and for cols.
+     */
+    private Vector<Pair> getSuccessors(Pair cell) {
+        Vector<Pair> succ = new Vector<>();
+        int row_offset[] = new int[]{-1, -1, -1, 0, 0, 1, 1, 1};
```

Comment:
You are initiating these two arrays every time you call getSuccessors method. You could make static final arrays and refer to them from this method. That is much more efficient.